libsolutil/JSON.h

```diff
@@ -23,15 +23,17 @@
 
 #pragma once
 
-#include <json/json.h>
+#include <json/json.hpp>
 
 #include <string>
 
+using Json = nlohmann::json;
```

Comment:
Do we really want this in a header?

Reply:
I don't think so, grandfathered in form @aarlt.

Reply:
Implying that all headers should use the full notation?

Reply:
Yeah, pretty much. I don't think we want to pollute the global namespace this way, do we?

Reply:
I suspect it will be very verbose to always write `nlohmann::json` instead of `Json`. It's also not exactly rolling of the tongue. 
Also, doesn't this conflict with the desire to abstract the library specific parts as mentioned here https://github.com/ethereum/solidity/pull/11967#discussion_r715772723

Reply:
And I guess it also means we'd need to use `using Json = nlohmann::json;` in all cpp files that use it then?